### PR TITLE
Added documentation for creating a test instance on App Engine

### PR DESCRIPTION
**Pull request by feihong**
This documentation is up-to-date as of April 15, 2016.

### CODE REVIEW

File: docs/source/topics/test-instance.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+======================================
+Creating a Test Instance on App Engine
+======================================
+
```
Comment (kumar303) [inline docs/source/topics/test-instance.rst]: Can you add a short summary here of why you would want to do this? Something like: "If you need to test out your changes in a real App Engine environment, here's how to deploy a test instance that is completely separate from the production site."

File: docs/source/topics/test-instance.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#. Go to the `API Manager credentials <https://console.cloud.google.com/apis/credentials>`_ page.
+#. Select Create credentials > Service account key.
+#. Select App Engine default service account, choose JSON, and hit Create. The newly-generated key will be automatically downloaded to your computer.
+#. Connect to your remote datastore using the `Remote API Shell <https://cloud.google.com/appengine/docs/python/tools/remoteapi#using_the_remote_api_shell>`_::
+
+     GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account_key.json remote_api_shell.py -s <your project id>.appspot.com
+
+#. To create a new user, run the following code in the shell:
+
+   .. code-block:: python
```
Comment (kumar303) [inline docs/source/topics/test-instance.rst]: This needs to start on a new line without preceding whitespace, like:

    .. code-block:: python

Comment (kumar303) [inline docs/source/topics/test-instance.rst]: well, actually, does it work like this? I guess you were trying to preserve the numbered bullet points.

**Comment by kumar303** [comment]
Looks good, thanks! Just some minor requests

**Comment by feihong** [comment]
@kumar303 Yeah, I have to indent the code block to preserve the numbered bullet points. Note that added a line to .gitignore in this latest commit.

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@
 .DS_Store
 *~
 *.mo
+docs/build
```
Comment (kumar303) [inline .gitignore]: thanks

**Comment by kumar303** [comment]
looks good, thanks!

**Pull request merged**